runelite-client/src/main/java/net/runelite/client/plugins/dailytaskindicators/DailyTasksPlugin.java

```diff
@@ -0,0 +1,173 @@
+import java.awt.Color;
+import javax.inject.Inject;
+import lombok.extern.slf4j.Slf4j;
+import net.runelite.api.ChatMessageType;
+import net.runelite.api.Client;
+import net.runelite.api.Varbits;
+import net.runelite.api.events.ConfigChanged;
+import net.runelite.api.events.GameTick;
+import net.runelite.api.events.VarbitChanged;
+import net.runelite.client.chat.ChatColor;
+import net.runelite.client.chat.ChatColorType;
+import net.runelite.client.chat.ChatMessageBuilder;
+import net.runelite.client.chat.ChatMessageManager;
+import net.runelite.client.chat.QueuedMessage;
+import net.runelite.client.config.ConfigManager;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+
+@PluginDescriptor(
+	name = "Daily Task Indicator"
+)
+@Slf4j
+public class DailyTasksPlugin extends Plugin
+{
+	@Inject
+	private Client client;
+
+	@Inject
+	private DailyTasksConfig config;
+
+	@Inject
+	private ChatMessageManager chatMessageManager;
+
+	private boolean canCollectHerbBox, canCollectStaves, canCollectEssence;
+	private boolean sentMessageThisSession;
+
+	@Provides
+	DailyTasksConfig provideConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(DailyTasksConfig.class);
+	}
+
+	@Override
+	protected void startUp() throws Exception
+	{
+		sentMessageThisSession = canCollectHerbBox = canCollectStaves = canCollectEssence = false;
+		cacheColors();
+	}
+
+	@Override
+	protected void shutDown() throws Exception
+	{
+		sentMessageThisSession = canCollectHerbBox = canCollectStaves = canCollectEssence = false;
+	}
+
+	@Subscribe
+	public void onConfigChanged(ConfigChanged event)
+	{
+		if (event.getGroup().equals("dailytaskindicators"))
+		{
+
+		}
+	}
+
+	@Subscribe
+	public void onVarbitChanged(VarbitChanged event)
+	{
+		if (!sentMessageThisSession)
+		{
+			if (config.showHerbBoxes())
+			{
+				checkCanCollectHerbBox(client.getSetting(Varbits.DAILY_HERB_BOX));
+			}
+			if (config.showStaves())
+			{
+				checkCanCollectStaves(client.getSetting(Varbits.DAILY_STAVES));
+			}
+			if (config.showEssence())
+			{
+				checkCanCollectEssence(client.getSetting(Varbits.DAILY_ESSENCE));
+			}
+		}
+	}
+
+	@Subscribe
+	public void onGameTick(GameTick event)
+	{
+		if (!sentMessageThisSession)
+		{
+			if (config.showHerbBoxes() && canCollectHerbBox)
+			{
+				sendChatMessage("You have herb boxes waiting to be collected at NMZ.");
+			}
+			else
+			{
+				System.out.println(client.getSetting(Varbits.DAILY_HERB_BOX));
+			}
+			if (config.showStaves() && canCollectStaves)
+			{
+				sendChatMessage("You have staves waiting to be collected from Zaff.");
+			}
+			else
+			{
+				System.out.println(client.getSetting(Varbits.DAILY_STAVES));
+			}
+			if (config.showEssence() && canCollectEssence)
+			{
+				sendChatMessage("You have pure essence waiting to be collected from Wizard Cromperty.");
+			}
+			else
+			{
+				System.out.println(client.getSetting(Varbits.DAILY_ESSENCE));
```

Comment:
Remove this, we have logger, probably also add some sensible message instead, or just completely remove the log